### PR TITLE
Lf 4015 hotfix routes

### PR DESCRIPTION
**Pull request by Duncan-Brain**
**Description**
Fixing finances routes.

Jira link:

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/util/siteMapConstants.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2023 LiteFarm.org
```
Comment (SayakaOno) [inline packages/webapp/src/util/siteMapConstants.ts:2]: 2024! 😆 
Thank you for working on this, I will start testing now!

**Comment by antsgar** [comment]
Looks good to me, at least on paper! I think we should merge it sooner rather than later to get more testing time on beta. Should we remove the deprecated routes file as part of this PR too?

**Review by antsgar — DISMISSED** [review]

File: packages/webapp/src/util/siteMapConstants.ts (hunk 1 above)
Comment (kathyavini) [reply packages/webapp/src/util/siteMapConstants.ts:2]: I'm surprised we didn't have more of that these last two weeks!

File: packages/webapp/src/routes/index.jsx [hunk 2]
```diff
@@ -901,7 +899,6 @@ const Routes = ({ isCompactSideMenu }) => {
             />
             <Route path="/finances" component={Finances} />
```
Comment (Duncan-Brain) [inline packages/webapp/src/routes/index.jsx:900]: @antsgar Just noticed.. is this missing a `/finances/* `like the other one in this file?

File: packages/webapp/src/routes/FinancesRoutes.jsx [hunk 3]
```diff
@@ -58,45 +81,45 @@ const EditCustomRevenue = React.lazy(() =>
...
+      path={createReadonlyCustomRevenueUrl(':revenue_type_id')}
       exact
       component={ReadOnlyCustomRevenue}
     />
     <Route
-      path="/finances/edit_custom_revenue/:revenue_type_id"
+      path={createEditCustomRevenueUrl(':revenue_type_id')}
       exact
       component={EditCustomRevenue}
     />
```
Comment (Duncan-Brain) [inline packages/webapp/src/routes/FinancesRoutes.jsx:125]: @antsgar The redirect below this line.. should it be redirecting from `/finances` to FINANCES_HOME_URL ?

**Comment by kathyavini** [comment]
> Should we remove the deprecated routes file as part of this PR too?

@antsgar I see it as deleted 🤔

**Review by kathyavini — APPROVED** [review]

File: packages/webapp/src/routes/FinancesRoutes.jsx (hunk 3 above)
Comment (antsgar) [reply packages/webapp/src/routes/FinancesRoutes.jsx:125]: I'm not sure honestly, I think I copied that from similar files

**Review by antsgar — APPROVED** [review]

**Pull request merged**